### DISCUSSION TITLE
Spellbook Moves to a new License

### DISCUSSION THREAD

**Discussion by augustog** [duneanalytics/spellbook#3452] [General]
Dear Wizards,

**TL;DR:** We are changing Spellbook to a BSL license. While this should not result in any practical change for most Spellbook contributors and Dune users/customers, we wanted to ensure you are aware. 

The [BSL license](https://github.com/duneanalytics/spellbook/blob/main/LICENSE) is the same license that Uniswap V3 was licensed under.

We’re doing this to ensure that 
- Spellbook contributors are properly attributed for their work
     - Currently, any third party can take any part of Spellbook and exploit it without reference to the community members doing the heavy lifting on spellbook

- Dune can continue to build and invest in our community data platform 
     - Spellbook is growing quickly and we are spending significant time and resources to expand and maintain it
     - As Dune remains committed to our mission of making crypto data accessible, we’ve drafted the most permissive license that still enables us to protect our and our community’s collective work

What the BSL license means in practice
- All your contributions (past and future) are yours.
- The Spellbook code will remain transparent and auditable.
- Usage of Spellbook code outside of Dune will still be allowed as long as it’s for internal or personal use

The license in a nutshell
- All of the source code of Spellbook can be shared freely and used for any purpose provided it does not expose the raw or aggregate output to third parties in an alternative data platform
- New Releases will be open sourced under GPLv3 after two years from the release date, ensuring that the community’s work becomes fully open source 
- Your contributions are yours, and you’re free to use them or reproduce them anywhere. 

As part of these changes, we have also added a [Contributor License Agreement (CLA) to spellbook](https://github.com/duneanalytics/spellbook/blob/main/CLA.md). This is meant to clarify and put in writing the generally understood premise of contributing to OSS, which is that contributors retain their ownership of their contributions but allow others to use it as part of this project. We are using the Apache Foundation CLA which is commonly used by leading OSS projects.

We are excited to keep making crypto data accessible together with all of you!
